lib/blurb/campaign.rb

```diff
@@ -1,29 +1,30 @@
 module Blurb
   class Campaign < BaseResource
-    def self.retrieve(campaign_id)
-      get_request("/v1/campaigns/#{campaign_id}")
+    def self.retrieve(campaign_id, campaign_type)
+      get_request("/v2/#{campaign_type}/campaigns/#{campaign_id}")
     end
 
     def self.retrieve_extended(campaign_id)
```

Comment:
You aren't passing `campaign_type` here, and it's used later
```suggestion
    def self.retrieve_extended(campaign_id, campaign_type)
```

Reply:
Good catch!  Thanks.